lib/tasks/server/middleware/history-support/index.js

```diff
@@ -29,7 +29,9 @@ HistorySupportAddon.prototype.serverMiddleware = function(config) {
 
         if (hasHTMLHeader && isForBaseURL && req.method === 'GET') {
           var assetPath = req.path.slice(baseURL.length);
-          if (!fs.existsSync(path.join(results.directory, assetPath))) {
+          var isFile = false;
+          try { isFile = fs.statSync(path.join(results.directory, assetPath)).isFile(); } catch (err) { }
+          if (!isFile) {
```

Comment:
Broccoli's middleware just calls `next()` instead of serving 404. So if possible, it might be nicer to rearrange the middleware stack so this check becomes unnecessary. I just didn't want to try doing this right now.